### DISCUSSION TITLE
is there a way to programatically create a new source item?

### DISCUSSION THREAD

**Discussion by endingwithali** [andreykaipov/goobs#155] [Q&A]
im trying to programmatically create a new source based off a image i programmatically generate. I see that apparently in older versions of goobs there was a createsource method in the sources client, but that method doesnt exist anymore. 

so, given a file location, how can i create a new image source?

TIA <3

**Comment by 3-io** [comment]
If you mean to add to the "Sources" list within OBS, you instead need to use inputs.

Using your example of creating an image source to add to a scene, and assuming that you have already connected the goobs client, you can achieve this with the following:

```go
client, _ := goobs.New("localhost:4455", goobs.WithPassword("hunter2"))

// Get the currently active scene
scene, _ := client.Scenes.GetCurrentProgramScene()

// Compose the new input
iParams := inputs.
	NewCreateInputParams().
	WithSceneName(scene.CurrentProgramSceneName).
	WithInputKind("image_source").
	WithInputName("New Image").
	WithInputSettings(map[string]interface{}{
		"file": "/path/to/your/image/file",
	})

// Send the input over to OBS
iRes, err := client.Inputs.CreateInput(iParams)
if err != nil {
	panic(err)
}
```

You should now see "New Image" in your OBS sources list.
